src/TTImp/Elab/RunElab.idr

```diff
@@ -51,6 +52,32 @@ Reflect NameInfo where
       = do nt <- reflect fc defs lhs env (nametype inf)
            appCon fc defs (reflectiontt "MkNameInfo") [nt]
 
+deepRefersTo : {auto c : Ref Ctxt Defs} ->
+               GlobalDef -> Core (List Name)
+deepRefersTo def = do
+  defs <- get Ctxt
+  map nub $ clos empty defs $ refs' defs def
+  -- we don't start with `[defs.fullname]` to distinguish between recursive and non-recursive definitions
+  where
+    refs' : Defs -> GlobalDef -> List Name
+    refs' defs def = do
+        let addition = case definition def of
+                         TCon {datacons, _} => datacons
```

Comment:
This was not a sublime, rather ridiculous. I first implemented it without this, and after that I though -- hmm, data types are *definitions* in a sence (there is a `GlobalDef` and `Def` corresponding to them). Then, I added this traversal. But now I understand that I meant another sort of definition, something like `IDef`, and those that are managed by `processDef`, i.e. *function definitions*, so this shouldn't be this way.

So, I renamed proposed function in the `Elaboration` interface to mention *functions* rather than *definitions* to not to use too overloaded word "definition", and rebased it.